### DISCUSSION TITLE
Compile for PowerPC Macs? (Leopard/Tiger)

### DISCUSSION THREAD

**Discussion by mzryz** [debauchee/barrier#1302] [Q&A] [answered]
Hi, I'd like to compile for PPC. I tried the most obvious/easy route and tried installing via Macports, but it said Intel x86 only. I was surprised as I have Barrier compiled for MIPS, so surely it must be possible to compile for PPC?

If it is possible, could someone provide me with the command line syntax I'd use for compiling, thank you!

**Comment by shymega** [comment, marked as answer]
We don't support PPC, and therefore I personally wouldn't be able to give any advice. Perhaps a good starting point would be to google 'cmake cross-compiling to MIPS' or something. But you'll need the QT5 libs on PPC, and Avahi etc. Probably more work than its worth.

On Fri, Oct 1, 2021, at 11:40 AM, mzryz wrote:
> Hi, I'd like to compile for PPC. I tried the most obvious/easy route 
> and tried installing via Macports, but it said Intel x86 only. I was 
> surprised as I have Barrier compiled for MIPS, so surely it must be 
> possible to compile for PPC?
>
> If it is possible, could someone provide me with the command line 
> syntax I'd use for compiling, thank you!
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/debauchee/barrier/discussions/1302>, or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AAKF2QECRSJYKY2ZWNGH3CDUEWF2DANCNFSM5FEMOJVA>.

-- 
Kind regards,
--
Dom Rodriguez (also known as shymega)

**Reply by mzryz** [reply]
Thank you, but I don't need barrier as a server, just the client is required. Which strips out almost all the difficult requirements. I just need 'barrierc'.

**Reply by barracuda156** [reply]
> We don't support PPC, and therefore I personally wouldn't be able to give any advice. Perhaps a good starting point would be to google 'cmake cross-compiling to MIPS' or something. But you'll need the QT5 libs on PPC, and Avahi etc. Probably more work than its worth.
> […](#)

`avahi` builds for PowerPC, however Qt5 does not, presently. (Wish someone can fix it.)

**Reply by shymega** [reply]
Yes, if Avahi does, then that removes one piece of the puzzle. If `barrierc` is only needed, then there's no need for Qt, and @mzryz can build Barrier. I am no longer involved with Barrier, so I can't help any further on this repository. I doubt there'll be any work to support Qt5 on PPC, sorry.

**Comment by barracuda156** [comment]
An obvious pick would be use a fallback version which worked on PPC with Qt4. Does anyone know which one was that? I can add it to Macports.
